tests/test_json.py

```diff
@@ -89,6 +89,14 @@ def test_register_from_layout():
     assert new_reg._layout_info.trap_ids == (1, 0)
 
 
+def test_register_numbered_keys():
+    reg = Register(dict(enumerate([(2, 3), (5, 1), (10, 0)])))
+    j = json.dumps(reg, cls=PulserEncoder)
+    decoded_reg = json.loads(j, cls=PulserDecoder)
+    assert reg == decoded_reg
+    assert all([type(i) == int for i in decoded_reg.qubit_ids])
```

Comment:
Can you maybe add a test with the `Register3D`too? And on that one, perhaps use an unconventional order for the indices and make sure that is respected